src/Transports.AspNetCore/Extensions/GraphQLBuilderUserContextExtensions.cs

```diff
@@ -80,6 +80,49 @@ public static IGraphQLBuilder AddUserContextBuilder<TUserContext>(this IGraphQLB
         return builder;
     }
 
+    /// <summary>
+    /// Configures a delegate to be used to create a user context for each GraphQL request.
+    /// <br/><br/>
+    /// Requires <see cref="IHttpContextAccessor"/> to be registered within the dependency injection framework
+    /// if calling <see cref="DocumentExecuter.ExecuteAsync(ExecutionOptions)"/> directly.
+    /// </summary>
+    public static IGraphQLBuilder AddUserContextBuilder<TUserContext>(this IGraphQLBuilder builder, Func<HttpContext, object?, TUserContext> creator)
+        where TUserContext : class, IDictionary<string, object?>
+    {
+        builder.Services.Register<IUserContextBuilder>(new UserContextBuilder<TUserContext>(creator ?? throw new ArgumentNullException(nameof(creator))));
+        builder.ConfigureExecutionOptions(options =>
+        {
+            if (options.UserContext == null || options.UserContext.Count == 0 && options.UserContext.GetType() == typeof(Dictionary<string, object>))
+            {
+                var requestServices = options.RequestServices ?? throw new MissingRequestServicesException();
+                var httpContext = requestServices.GetRequiredService<IHttpContextAccessor>().HttpContext!;
+                options.UserContext = creator(httpContext, null);
```

Comment:
I would pull context builder here and in another method below from DI as `  public static IGraphQLBuilder AddUserContextBuilder<TUserContextBuilder>(this IGraphQLBuilder builder)` does. I have a suspicion that in the case of subsequent registrations of the context builder a situation may arise when options are still configured with the "old" `creator` delegate from closure.

Reply:
I rewrote the methods to conform to your suggestion. Please review.